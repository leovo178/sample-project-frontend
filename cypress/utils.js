/* eslint-disable no-undef */
const generateRandomNumber = (min, max) => {
	min = Math.ceil(min)
	max = Math.floor(max)
	return Math.floor(Math.random() * (max - min)) + min
}

const clickItemByTestId = testId =>
	cy
		.findByTestId(testId)
		.should('exist')
		.click()

const mockMessage = 'mockmessage'

export { generateRandomNumber, clickItemByTestId, mockMessage }
