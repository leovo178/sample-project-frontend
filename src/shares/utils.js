const AUTH_TOKEN = 'AUTH_TOKEN'

const getToken = () => localStorage.getItem(AUTH_TOKEN)
const setToken = token => localStorage.setItem(AUTH_TOKEN, token)
const deleteToken = () => localStorage.removeItem(AUTH_TOKEN)

const validateEmail = email => {
	const emailRegex = /^[a-z][a-z0-9_\.]{4,32}@[a-z0-9]{2,}(\.[a-z0-9]{2,4}){1,2}$/gm
	return email.match(emailRegex)
}

const validateName = name => {
	const nameRegex = /^[a-zA-Z0-9]{3,30}$/
	return name.match(nameRegex)
}

const validatePassword = password => {
	const passwordRegex = /^[a-zA-Z0-9]{3,}$/
	return password.match(passwordRegex)
}

export {
	getToken,
	setToken,
	deleteToken,
	validateEmail,
	validateName,
	validatePassword,
}
