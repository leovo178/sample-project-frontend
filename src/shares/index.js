import { getToken, setToken, deleteToken } from './utils'

export { getToken, setToken, deleteToken }
