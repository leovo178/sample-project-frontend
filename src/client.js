import { ApolloClient } from 'apollo-client'
import { createUploadLink } from 'apollo-upload-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { setContext } from 'apollo-link-context'

import { getToken } from './shares/utils'
import { APOLLO_SERVER } from './configs.local'

const typeDefs = {}

const resolvers = {
	Mutation: {
		setUserSearchValue: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValue: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUser: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUser,
				},
			})
			return selectedUser
		},
		setUploadedFile: (_, { file }, { cache }) => {
			if (file) {
				cache.writeData({
					data: {
						file,
					},
				})
			}
			return file
		},
		setUserSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValueOfMain: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUserOfMain: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUserOfMain: selectedUser,
				},
			})
			return selectedUser
		},

		setMessageSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					messageSearchValueOfMessage: searchValue,
				},
			})
			return searchValue
		},
		setMessageCreateValueOfMain: (_, { createValue }, { cache }) => {
			cache.writeData({
				data: {
					messageCreateValueOfMessage: createValue,
				},
			})
			return createValue
		},
		resetCache: (_, { data }, { cache }) => {
			cache.writeData({
				data,
			})
			return data
		},
	},
}

const httpLink = createUploadLink({
	uri: APOLLO_SERVER,
	credentials: 'same-origin',
})

const authLink = setContext((_, { headers }) => {
	const token = getToken()
	return {
		headers: {
			...headers,
			authorization: token ? `Bearer ${token}` : '',
		},
	}
})

const cache = new InMemoryCache()

const client = new ApolloClient({
	cache,
	link: authLink.concat(httpLink),
	typeDefs,
	resolvers,
})

const initialState = {
	userSearchValue: '',
	messageCreateValueOfMessage: '',
	messageSearchValueOfMessage: '',
	userSearchValueOfMain: '',
	selectedUser: {
		id: '',
		name: '',
		email: '',
		__typename: 'User',
	},
	file: {
		filename: '',
		link: '',
		__typename: 'File',
	},
	selectedUserOfMain: {
		id: '',
		name: '',
		email: '',
		__typename: 'UserOfMain',
	},
}

cache.writeData({
	data: initialState,
})

export { client, initialState }
