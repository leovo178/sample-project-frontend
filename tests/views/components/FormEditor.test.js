import React from 'react'

import { createMockClient } from 'mock-apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { ApolloProvider } from '@apollo/react-hooks'

import { render, fireEvent, cleanup } from '@testing-library/react'

import { FormEditor } from '@views_components'

import { renderWithRouter } from '@tests/shares/utils'

import { initialState, resolvers } from '@src/client'

describe('FormEditor', () => {
	const cache = new InMemoryCache()
	cache.writeData({
		data: initialState,
	})
	const mockClient = createMockClient({
		cache,
		resolvers,
	})

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', () => {
		const { container } = renderWithRouter(
			<ApolloProvider client={mockClient}>
				<FormEditor />
			</ApolloProvider>
		)

		expect(container).toMatchSnapshot()
	})

	it('should render Sign up form without crashing', () => {
		const {
			getByText,
			getByTestId,
			queryAllByLabelText,
			findAllByLabelText,
		} = renderWithRouter(
			<ApolloProvider client={mockClient}>
				<FormEditor />
			</ApolloProvider>
		)

		console.log(getByTestId('form-editor-email'))

		expect(getByText('Sign up')).toBeTruthy()
		// expect(queryByLabelText('EMAIL')).toBeTruthy()
		// expect(getByText('NAME')).toBeTruthy()
		// expect(getByText('PASSWORD')).toBeTruthy()
		// expect(getByText('PASSWORD CONFIRM')).toBeTruthy()
		// expect(getByText('Register')).toBeTruthy()
	})

	// it('should call correctly onAgree function when click Yes button', () => {
	// 	const props = { ...mockProps, open: true }
	// 	const { getByText } = render(<FormEditor {...props} />)

	// 	const yesButton = getByText('Yes')

	// 	fireEvent.click(yesButton)

	// 	expect(props.onAgree).toHaveBeenCalled()
	// })
})
