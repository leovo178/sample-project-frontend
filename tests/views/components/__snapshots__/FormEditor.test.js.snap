// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormEditor should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-8 makeStyles-root-1"
  >
    <h5
      class="MuiTypography-root makeStyles-formtitle-2 MuiTypography-h5"
    >
      Sign up
    </h5>
    <div
      class="makeStyles-formcontent-4"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-forminput-3"
        data-testid="email"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          EMAIL
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-97 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-99"
            >
              <span>
                EMAIL
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-forminput-3"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          NAME
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="true"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-97 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-99"
            >
              <span>
                NAME
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-forminput-3"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="formeditor-password"
          id="formeditor-password-label"
        >
          PASSWORD
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="true"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="formeditor-password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-97 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-99"
            >
              <span>
                PASSWORD
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-forminput-3"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="formeditor-password-confirm"
          id="formeditor-password-confirm-label"
        >
          PASSWORD CONFIRM
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="true"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="formeditor-password-confirm"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-97 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-99"
            >
              <span>
                PASSWORD CONFIRM
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-formbuttons-5"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formbutton-6 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        data-cy="submit-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Register
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formbutton-6 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
